src/plugins/data/server/search/strategies/ese_search/ese_search_strategy.ts

```diff
@@ -170,7 +171,7 @@ export const enhancedEsSearchStrategyProvider = (
      */
     search: (request, options: IAsyncSearchOptions, deps) => {
       logger.debug(`search ${JSON.stringify(request.params) || request.id}`);
-      if (request.indexType && request.indexType !== 'rollup') {
+      if (request.indexType && request.indexType !== 'rollup' && request.indexType !== ESQL_TYPE) {
```

Comment:
@lukasolson suggested taking another approach, but it would get rid of the error if there was a bad indexType provided. I want to make sure thats what he wants before implementing it.

Reply:
I'd suggest to take care of the ESQL_TYPE in a follow up, because it's not a requirement in this PR, right? And it's cleaner to do it separately